### ISSUE TITLE
Make hamcrest work with LocalHere

### ISSUE THREAD

**Issue by VDFaller** [sassoftware/jsl-hamcrest#129]
I'm trying to write a test similar to 
```jsl
uttest("some case", "localhere", 
  Expr(
    LocalHere(
      ::utassertthat(Expr(1+2), ::utgreaterthan(2), "Simple")
    ) 
  )
);
```

And get the following
```
FAILURE  _utAssertThatNS:test expr /*###*/ test does not throw outside of ... (some case ⮚ localhere ⮚ Test)
    Actual: threw "Error: The namespace "_utAssertThatNS" is not defined. 1 time. _utAssertThatNS:test expr /*###*/"
    Expected: test does not throw outside of an assertion
```

I assume this would be more than just a simple `substitute(txt, "_ut", "::_ut", "ut", "::ut")`

The use case is that I'm running something two different ways, and I'm trying to not have the second time overwrite the first.  It sets some here variables.

**Comment by emccorkle** [comment]
The `_ut` ones are locals so those wouldn't need to be modified. I *was* able to get the body of your test to run by replacing `ut *` with `::ut *` in Hamcrest as errors popped up (parts of `Core/Core`, `Core/Utils`, and `Reporters/StreamingLogReporter` did it). And by changing `ut global reporter:msg` to `::ut global reporter << msg`. However, note that when I run the test above, I start with an error about `ut move to anonymous namespace` instead of `_utAssertThatNS`. Here is the script I was using for that:
```
Names Default To Here(0);
Include("Source\All.jsl");
::ut global reporter = ::ut streaming log reporter();
Local Here(
    ::ut assert that(Expr(1+2), ::ut greater than(3), "Simple")
) 
```
I think this would be pretty easy to do codebase-wide and test. The sticking point is likely `ut test`. It expects `ut assert that` to be used unqualified within it because it wraps `ut assert that` so that the test labels can be augmented and the aggregate test status reported. I think we could probably keep the unqualified assumption and modify how the wrapping is done so it works if `Names Default To Here` is off or on.

**Comment by VDFaller** [comment]
Is that the reason when I use ::utassertthat() inside of an unscoped uttest that it just goes to "DEFAULT" for the case?

**Comment by emccorkle** [comment]
Yes, that is correct. That code makes use of `ut test register assertion` and `ut test assertion decorator` here https://github.com/sassoftware/jsl-hamcrest/blob/master/Source/Core/TestCase.jsl#L687-L762.
